added item drop magnet physics

NathanHinthorne opened the pull request.

NathanHinthorne committed:
better eel behavior

NathanHinthorne committed:
item drop magnet physics

NathanHinthorne committed:
Merge branch 'merge' into main

NathanHinthorne committed:
new eel sprites

NathanHinthorne committed:
Merge branch 'main' of https://github.com/NathanHinthorne/good-bad-chad-nathan

calebkrauter commented on js/classes/entities/midground/drops/ammo_drop.js:
```diff
@@ -51,17 +52,28 @@ class AmmoDrop {
         if (this.hasGravity) {
 
-            this.yVelocity += PHYSICS.GRAVITY_ACC * GAME.clockTick;
+            // move toward player like a magnet when close enough
```
Neat!

calebkrauter commented on js/classes/entities/midground/enemies/eelORIGINAL.js:
What is eel original? I am confused, why are there two eel classes? I'd suggest a refactor if these eel classes are both in use. If they are not in use perhaps we could delete one of them?

NathanHinthorne replied:
I meant to get rid of eel original. I believe in my latest commit I did end up getting rid of it.

calebkrauter commented on js/classes/entities/midground/enemies/flying_enemy_base.js:
```diff
@@ -123,4 +134,19 @@ class FlyingEnemyBase {
         return [new Vector(0, 0), new Vector(200, 0), new Vector(0, 0)];
     }
 
+    static get JUMP_SMALL() {
```
What is the purpose of these jump functions? Do flying enemies jump?

NathanHinthorne replied:
yeah, it's kind of strange, but they do jump. GroundEnemyBase doesn't support jumping to that extent, so I need to find a quick work-around.

calebkrauter commented on js/classes/entities/midground/enemies/slime.js:
```diff
@@ -143,6 +143,7 @@ class Slime {
         this.base.update();
 
         const deathAnim = this.animations[this.base.getFacing()]["dying"];
+        console.log("deathAnim: ", deathAnim);
```
Is this console.log necessary?

NathanHinthorne replied:
I'll get rid of it.

calebkrauter commented on js/classes/entities/midground/weapon/sword.js:
```diff
@@ -11,10 +11,18 @@ class Sword {
         this.lastAttack = 0;
         this.damage = Sword.LEVEL_1_DAMAGE;
+        
+        if (swordLevel === 1) {
```
What impacts the sword damage level?

NathanHinthorne replied:
The sword damage level will change when you upgrade your sword at the blacksmith.

calebkrauter commented on js/classes/entities/midground/weapon/sword.js:
```diff
@@ -49,6 +57,16 @@ class Sword {
     };
 
+    upgrade() {
+        if (this.damage === Sword.LEVEL_1_DAMAGE) {
```
Are you adding the upgrade system we had talked about originally?

NathanHinthorne replied:
You bet!

calebkrauter committed:
Update main.js

Looks like you neglected to make sure the game starts at the village? I'm suggesting we put this back.

calebkrauter commented on js/scripts/main.js:
```diff
@@ -72,7 +72,7 @@ CANVAS.onfocus = () => {
 // (3) Set the current ZONE to be the first one we encounter - village.main.
 
-let ZONE = Zone.getZones().village.main;
+let ZONE = Zone.getZones().river.river1;
```
This should be village.main if I'm not mistaken.

NathanHinthorne replied:
Thanks! I'll change it.

calebkrauter commented on js/scripts/util.js:
```diff
@@ -125,7 +125,8 @@ const SFX = {
     WATER_BALLOON: {path: "./sfx/water_balloon.wav", volume: 0.4},
-    LIFE_UP: {path: "./sfx/life_up.wav", volume: 0.4},
+    LIFE_UP: { path: "./sfx/life_up.wav", volume: 0.4 },
+    SWORD_UPGRADE: { path: "./sfx/sword_upgrade.mp3", volume: 0.4 },
```
Would you be interested in updating the formatting so that it's all consistent in util.js? I'm referring to having a space before and after the data in the object next to the curly brackets. I don't care if we do or don't have a space, I am just not a fan of the inconsistency.

NathanHinthorne replied:
Sure, I can do that.

calebkrauter commented on js/scripts/zone_loaders/river_zone_loaders.js:
```diff
@@ -88,9 +89,9 @@ const loadRiver = () => {
-    GAME.addEntity(new Eel(Vector.blockToWorldSpace(new Vector(8.5, 29)), 500, 0, 8));
-    GAME.addEntity(new Eel(Vector.blockToWorldSpace(new Vector(20.5, 29)), 700, 30, 10));
-    GAME.addEntity(new Eel(Vector.blockToWorldSpace(new Vector(25.5, 29)), 700, -30, 12));
+    GAME.addEntity(new Eel(Vector.blockToWorldSpace(new Vector(8.5, 29)), FlyingEnemyBase.JUMP_SMALL, 450));
```
I'm not sure if JUMP makes sense here, is the enemy jumping or flying up?

NathanHinthorne replied:
It's the enemy flying up. It's a work-around.

calebkrauter commented on js/scripts/zone_loaders/village_zone_loaders.js:
```diff
@@ -342,15 +343,15 @@ const loadVillageMain = () => {
-        //     [new Vector(300, 100), new Vector(-300, -100), new Vector(0, 0)]
-        // ));
+        GAME.addEntity(new OculiBot(Vector.blockToWorldSpace(
+            new Vector(93, aboveGroundLevel - 4)),
```
Neat!

calebkrauter reviewed (approved):
Looks great!

NathanHinthorne committed:
blacksmith with proper items

NathanHinthorne committed:
Merge branch 'main' of https://github.com/NathanHinthorne/good-bad-chad-nathan

NathanHinthorne committed:
spawnpoints
